app/services/achievement-factory.js

```diff
@@ -0,0 +1,291 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+import { computed, defineProperty } from '@ember/object';
+import { gte, lte } from '@ember/object/computed';
+import upgrade from 'incremental-empire/utils/upgrade';
+
+export default Service.extend({
+  store: service(),
+  achievementPlan: undefined, // Map that will hold all the achievement definitions
+
+  init() {
+    this._super(...arguments)
+    this.achievementPlan = new Map()
+    this.achievementPlan.set('Reach turn 10', (achievement) => {
+      achievement.setProperties({
+        description: 'You can now assign template points to restart with more population',
+        templatePoint: 1,
+      })
+      defineProperty(achievement, 'condition', gte('game.empire.turn', 10))
```

Comment:
Make a helper `this._condition(achievement, computed_thingy)` (or just `_condition`) to avoid copy-pasting this field everywhere.